src/components/GameBoard.js

```diff
@@ -0,0 +1,105 @@
+import React, {useState} from "react";
+import Square from "./Square";
+import GameModeForm from "./GameModeForm";
+
+const GameBoard = ({endGame}) => {
+    const [gameBoard,setGameBoard] = useState([[]])
+    const [flagsCount, setFlagsCount] = useState(-1);
+    const [matrixSize, setMatrixSize] = useState({rows:0,cols:0});
```

Comment:
if `createGameMatrix()` accepts `rows,cols,mines`.
 Why not rename if to `const [gameSettings, setGameSettings] = useState({rows:0,cols:0,mines:0});`?